src/mono/wasm/host/DevServer/DebugProxyLauncher.cs

```diff
@@ -0,0 +1,216 @@
+        var ownerPid = Environment.ProcessId;
+
+        var processStartInfo = new ProcessStartInfo
+        {
+            FileName = "dotnet" + (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : ""),
+            Arguments = $"exec \"{executablePath}\" --OwnerPid {ownerPid} --DevToolsUrl {devToolsHost} --IsFirefoxDebugging {isFirefox} --FirefoxProxyPort 6001",
+            UseShellExecute = false,
+            RedirectStandardOutput = true,
+            RedirectStandardError = true,
+        };
+        RemoveUnwantedEnvironmentVariables(processStartInfo.Environment);
+
+        var debugProxyProcess = Process.Start(processStartInfo);
+        if (debugProxyProcess is null)
+        {
+            tcs.TrySetException(new InvalidOperationException("Unable to start debug proxy process."));
+        }
+        else
+        {
+            PassThroughConsoleOutput(debugProxyProcess);
+            CompleteTaskWhenServerIsReady(debugProxyProcess, isFirefox, tcs);
+
+            new CancellationTokenSource(DebugProxyLaunchTimeout).Token.Register(() =>
+            {
+                tcs.TrySetException(new TimeoutException($"Failed to start the debug proxy within the timeout period of {DebugProxyLaunchTimeout.TotalSeconds} seconds."));
+            });
+        }
+
+        return await tcs.Task;
+    }
+
+    private static void RemoveUnwantedEnvironmentVariables(IDictionary<string, string?> environment)
+    {
+        // Generally we expect to pass through most environment variables, since dotnet might
+        // need them for arbitrary reasons to function correctly. However, we specifically don't
+        // want to pass through any ASP.NET Core hosting related ones, since the child process
+        // shouldn't be trying to use the same port numbers, etc. In particular we need to break
+        // the association with IISExpress and the MS-ASPNETCORE-TOKEN check.
+        // For more context on this, see https://github.com/dotnet/aspnetcore/issues/20308.
+        var keysToRemove = environment.Keys.Where(key => key.StartsWith("ASPNETCORE_", StringComparison.Ordinal)).ToList();
+        foreach (var key in keysToRemove)
+        {
+            environment.Remove(key);
+        }
+    }
+
+    [UnconditionalSuppressMessage("SingleFile", "IL3000:Avoid accessing Assembly file path when publishing as a single file", Justification = "Not published as a single file")]
+    private static string LocateDebugProxyExecutable(IWebHostEnvironment environment)
+    {
+        if (string.IsNullOrEmpty(environment.ApplicationName))
+        {
+            throw new InvalidOperationException("IWebHostEnvironment.ApplicationName is required to be set in order to start the debug proxy.");
+        }
+        var assembly = Assembly.Load(environment.ApplicationName);
+        var debugProxyPath = Path.Combine(
+            Path.GetDirectoryName(assembly.Location)!,
+            "BlazorDebugProxy",
```

Comment:
`BlazorDebugProxy` do we have such thing ?

Reply:
Fixed